hal/test-headers/ticker_api_tests.h

```diff
@@ -0,0 +1,138 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017-2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** \addtogroup hal_specification */
+/** @{*/
+
+#ifndef MBED_US_TICKER_API_TESTS_H
+#define MBED_US_TICKER_API_TESTS_H
+
+#include "device.h"
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * \defgroup hal_specification_ticker Ticker
+ * Ticker specification and tests
+ *
+ * To run the microsecond ticker hal tests use the command:
+ *
+ *     mbed test -t <toolchain> -m <target> -n tests-mbed_hal-us_lp_ticker*
+ *
+ * # Defined behavior
+ * * The function ticker_init is safe to call repeatedly - Verified by test ::ticker_init_test
+ * * The function ticker_init resets the internal count and disables the ticker interrupt - Verified by test ::ticker_init_test
+ * * Ticker frequency is non-zero and counter is at least 8 bits - Verified by ::ticker_info_test
+ * * The ticker rolls over at (1 << bits) and continues counting starting from 0 - Verified by ::ticker_overflow_test
+ * * The ticker counts at the specified frequency +- 10% - Verified by ::ticker_frequency_test
+ * * The ticker increments by 1 each tick - Verified by ::ticker_increment_test
+ * * The ticker interrupt fires only when the ticker times increments to or past the value set by ticker_set_interrupt.
```

Comment:
@c1728p9 Here we need to test two cases:
1. an interrupt is not triggered when ticker_set_interrupt is called with a time from the past.
2. an interrupt is triggered in the right time.

We have noticed that on some platforms ticker interrupt is periodic. For example on K64F. Is this a problem? 
Maybe we need additional case to verify that interrupt is not periodic?

Reply:
> We have noticed that on some platforms ticker interrupt is periodic. For example on K64F. Is this a problem?

Periodic as ? set once and it reoccurs? or ?

Reply:
Yes, it reoccurs. If we call:
intf->set_interrupt(intf->read() + TICKER_INT_VAL);

then IRQ handler will be executed more than once, in the following moments:
intf->read() + TICKER_INT_VAL
intf->read() + 2*TICKER_INT_VAL
intf->read() + 3*TICKER_INT_VAL
etc.

Reply:
That should not happen, pls report it

Reply:
Issue has been reported:
https://github.com/ARMmbed/mbed-os/issues/5279